src/scripts/main.js

```diff
@@ -1 +1,61 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', () => {
+    resolve('First promise was resolved');
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', (e) => {
+    resolve('Second promise was resolved');
+  });
+
+  document.addEventListener('contextmenu', (e) => {
+    resolve('Second promise was resolved');
+  });
+});
+
+const thirdPromise = new Promise((resolve) => {
+  let leftClick = false;
+  let rightClick = false;
+
+  const handleClick = (e) => {
+    if (e.button === 0) {
+      leftClick = true;
+    } else if (e.button === 2) {
+      rightClick = true;
+    }
+
+    if (leftClick && rightClick) {
+      resolve('Third promise was resolved');
+    }
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
```

Comment:
The third promise resolves when both left and right clicks have occurred. However, it will resolve multiple times if the user continues to click both buttons. Consider removing the event listeners after the promise resolves to prevent multiple resolutions.